packages/block-library/src/post-featured-image/edit.js

```diff
@@ -172,11 +178,23 @@ export default function PostFeaturedImageEdit( {
 		if ( value?.id ) {
 			setFeaturedImage( value.id );
 		}
+
+		if ( value?.url && isBlobURL( value.url ) ) {
+			setTemporaryURL( value.url );
+		}
 	};
 
+	// Reset temporary url when media is available.
+	useEffect( () => {
+		if ( media && temporaryURL ) {
+			setTemporaryURL();
+		}
+	}, [ media, temporaryURL ] );
+
```

Comment:
I think we can ditch this effect and reset `setTemporaryURL` in `onSelectImage` when the feature image ID is set.

Reply:
I did the same thing before, however it takes time to set `media` after calling `onSelectImage`, and at that time, the placeholder is displayed. It causes flashing. So, I have to clear `temporaryURL` after media is set.

Reply:
That makes sense. Let's use the derived `mediaUrl` string value for the condition instead of the `media` object.